### PR TITLE
Fixes

### LINKED ISSUE #2003: Enchanting table bugs
Im using minecraft 1.8.4 and the newest rev from MCServer
1) place a enchanting table (with bookshelfs around it)
2) enchant someting (in creative mode) without placing lapis in the slot
3) open/close the enchanting gui with random items in the hotpar multiple times
4) move items around in the inventory in the enchanting gui
Sometime an item dissapears, sometime it moves into the laips slot and sometimes an item(stack) gets duplicated

### LINKED ISSUE #2041: can't put a pumpkin on my head
To reproduce
-Try to use a pumpkin as helmet
-MCServer wont let you do that, it will remove the pumpkin and spawn it as pickup.

### LINKED ISSUE #2052: Picked up items go to hotbar
When picking up items while you already have that item somewhere in your inventory the item still goes to the hotbar.

### PR DESCRIPTION
**Pull request by tigerw**
* Fixes #2052
* Fixes #2003
* Fixes #2041

### CODE REVIEW

File: src/Items/ItemBucket.h [hunk 1]
```diff
@@ -96,7 +96,7 @@ class cItemBucketHandler :
 				ASSERT(!"Inventory bucket mismatch");
 				return true;
 			}
-			if (a_Player->GetInventory().AddItem(cItem(NewItem), true, true) != 1)
+			if (a_Player->GetInventory().AddItem(cItem(NewItem)) != 1)
```
Comment (bearbin) [inline src/Items/ItemBucket.h]: Why are both "true"s being removed here, rather than only one?

File: src/UI/SlotArea.cpp [hunk 2]
```diff
@@ -2154,7 +2154,7 @@ bool cSlotAreaArmor::CanPlaceArmorInSlot(int a_SlotNum, const cItem & a_Item)
 {
 	switch (a_SlotNum)
 	{
-		case 0:  return ItemCategory::IsHelmet    (a_Item.m_ItemType);
+		case 0:  return (ItemCategory::IsHelmet(a_Item.m_ItemType) || (a_Item.m_ItemType == E_BLOCK_PUMPKIN));
 		case 1:  return ItemCategory::IsChestPlate(a_Item.m_ItemType);
 		case 2:  return ItemCategory::IsLeggings  (a_Item.m_ItemType);
 		case 3:  return ItemCategory::IsBoots     (a_Item.m_ItemType);
```
Comment (bearbin) [inline src/UI/SlotArea.cpp]: The indentation here looks a bit messed up. Could you fix it?

File: src/Items/ItemBucket.h (hunk 1 above)
Comment (tigerw) [inline src/Items/ItemBucket.h]: As one of them is defaulted to true.

File: src/UI/SlotArea.cpp (hunk 2 above)
Comment (SphinxC0re) [inline src/UI/SlotArea.cpp]: Why? is there another method of checking this? Also, do plugins have the ability to override this? Like for wearing a stone block?

**Comment by tigerw** [comment]
Merge?

**Pull request merged**